### ISSUE TITLE
REST OPTIONS method request is blocked by webpack-dev-server logic

### ISSUE THREAD

**Issue by sek1973** [angular/angular-cli#23859]
### Command

serve

### Is this a regression?

- [X] Yes, this behavior used to work in the previous version

### Description

Each OPTIONS request is automatically replaced with the 204 code type response, regardless the content and how the application intends to handle it.
OPTIONS type request is compliant with specification:
[https://developer.mozilla.org/en-US/docs/Glossary/Preflight_request](url)
and should be handled respectively. 

This refers to [https://github.com/webpack/webpack-dev-server/issues/4551](url) and partially to #23857

### Minimal Reproduction

Any application using proxy functionality and sending OPTIONS requests.
Please find a basic project to reproduce this issue here: [https://github.com/sek1973/options-issue](url).

### Exception or Error

```text
Depending on the application implementation it may cause unexpected http response for OPTIONS requests.
```

### Your Environment

```text
Angular CLI: 14.2.1
Node: 16.17.0
Package Manager: npm 8.18.0
OS: win32 x64

Angular: 14.2.0
... animations, common, compiler, compiler-cli, core, forms
... platform-browser, platform-browser-dynamic, router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1402.1
@angular-devkit/build-angular   14.2.1
@angular-devkit/core            14.2.1
@angular-devkit/schematics      14.2.1
@angular/cli                    14.2.1
@schematics/angular             14.2.1
rxjs                            7.5.6
typescript                      4.7.4
```

### Anything else relevant?

@angular-devkit/build-angular@14.2.1 uses webpack-dev-server@4.10.0 internally.
Commenting out or removing following code in (node_modules\webpack-dev-server\lib\Server.js) seems to fix this issue:
```JS
{
/**
 *
 * @param {Request} req
 * @param {Response} res
 * @param {NextFunction} next
 * @returns {void}
 *
 */
const optionsRequestResponseMiddleware = (req, res, next) => {
  if (req.method === "OPTIONS") {
    res.statusCode = 204;
    res.setHeader("Content-Length", "0");
    res.end();
    return;
  }
  next();
};

middlewares.push({
  name: "options-middleware",
  path: "*",
  middleware: optionsRequestResponseMiddleware,
});
}
```

**Comment by alan-agius4** [comment]
Duplicate of https://github.com/angular/angular-cli/issues/23857
